File: test/basic.c
#include <stdio.h>
#include <stdlib.h>

#include <mpi.h>

int main(int argc, char **argv)
{
   printf("going to init MPI\n");

   if(MPI_Init(&argc,&argv)) {
      perror("could not init MPI\n");
      return EXIT_FAILURE;
   }
   printf("MPI Init OK\n");

   if(MPI_Finalize()) {
      perror("could not close MPI\n");
      return EXIT_FAILURE;
   }

   printf("MPI closed\n");

   return EXIT_SUCCESS;
}

File: test/basic.f90
program basic

use mpi
use, intrinsic :: iso_fortran_env, only: stderr=>error_unit, compiler_version

implicit none (type, external)

integer :: ierr

print '(a)', "going to init MPI"

call MPI_INIT(ierr)
if (ierr /= MPI_SUCCESS) then
  write(stderr,'(a,i0)') "MPI_INIT failed with error code", ierr
  error stop
endif

print '(a)', "MPI Init OK"

call MPI_FINALIZE(ierr)
if (ierr /= MPI_SUCCESS) then
  write(stderr,'(a,i0)') "MPI_FINALIZE failed with error code", ierr
  error stop
endif

print '(a)', "MPI closed"

end program
